centreon-dsm/www/modules/centreon-dsm/core/configuration/services/DB-Func.php

```diff
@@ -224,35 +271,40 @@ function insertPoolInDB($ret = array())
 
 /**
  *
- * Check Pool Existance
- * @param $pool_name
+ * Check Pool Existence
+ * @param string $pool_name The pool name to check
+ * @return int 0 if the pool does not exist, 1 if it does
  */
 function testPoolExistence($pool_name)
 {
     global $pearDB;
 
-    $dbResult = $pearDB->query("SELECT * FROM `mod_dsm_pool` WHERE `pool_name` = '" . $pool_name . "'");
-    if ($dbResult->rowCount() == 0) {
-        return 0;
-    } else {
-        return 1;
-    }
+    $stmt = $pearDB->prepareQuery("SELECT * FROM `mod_dsm_pool` WHERE `pool_name` = :pool_name");
+    $pearDB->executePreparedQuery($stmt, [':pool_name' => [$pool_name, PDO::PARAM_STR]], true);
+
+    $exists = $stmt->rowCount() == 0 ? 0 : 1;
```

Comment:
COUNT(*) is faster than rowCount() for your information, so it's better practice. And you use a native method of PDO do you must to track the PDOException to a have a log. It's better to use a COUNT(") and a CentreonDb::fetch() :)